cmd/tuple/write_test.go

```diff
@@ -223,3 +224,132 @@ func TestParseTuplesFileData(t *testing.T) { //nolint:funlen
 		})
 	}
 }
+
+var writeTuplesTests = []struct {
+	name               string
+	file               string
+	hideImportedTuples bool
+	expectedStats      ImportStats
+	expectedOutput     string
+	expectedError      string
+}{
+	{
+		name:               "shows successful tuples and counts when hide-imported-tuples is false",
+		file:               "testdata/tuples.json",
+		hideImportedTuples: false,
+		expectedStats: ImportStats{
+			TotalTuples:      3,
+			SuccessfulTuples: 3,
+			FailedTuples:     0,
+		},
+		expectedOutput: `{
+						"successful": [
+								{
+										"user": "user:anne",
+										"relation": "owner",
+										"object": "folder:product"
+								},
+								{
+										"user": "folder:product",
+										"relation": "parent",
+										"object": "folder:product-2021"
+								},
+								{
+										"user": "user:beth",
+										"relation": "viewer",
+										"object": "folder:product-2021"
+								}
+						],
+						"total_count": 3,
+						"successful_count": 3,
+						"failed_count": 0
+				}`,
+	},
+	{
+		name:               "shows only counts when hide-imported-tuples is true",
+		file:               "testdata/tuples.json",
+		hideImportedTuples: true,
+		expectedStats: ImportStats{
+			TotalTuples:      3,
+			SuccessfulTuples: 3,
+			FailedTuples:     0,
+		},
+		expectedOutput: `{
+						"total_count": 3,
+						"successful_count": 3,
+						"failed_count": 0
+				}`,
+	},
+	{
+		name:               "shows failed tuples and all counts",
+		file:               "testdata/tuples_with_errors.json",
+		hideImportedTuples: true,
+		expectedStats: ImportStats{
+			TotalTuples:      3,
+			SuccessfulTuples: 1,
+			FailedTuples:     2,
+		},
+		expectedOutput: `{
+						"failed": [
+								{
+										"user": "invalid:user",
+										"relation": "invalid",
+										"object": "invalid:object"
+								},
+								{
+										"user": "another:invalid",
+										"relation": "invalid",
+										"object": "invalid:object"
+								}
+						],
+						"total_count": 3,
+						"successful_count": 1,
+						"failed_count": 2
+				}`,
+	},
+}
+
+func TestWriteTuplesWithImportStats(t *testing.T) {
```

Comment:
It looks like we're mostly recreating the logic for the output in this test and not running the `fga tuple write` code? Given the mocking we would have to perform would [extending the integration tests](https://github.com/openfga/cli/blob/main/tests/tuple-test-cases.yaml) potentially be an easier way to test this?

Reply:
Resolved in c9d5dbd

Reply:
@ewanharris can you help me with that? it seems test are still failing ;_;